File: apps/worker/fielddesk_worker/db_queries/transcripts.py
from __future__ import annotations

from typing import Any
from uuid import UUID

from fielddesk_worker.db_queries._helpers import returned_id


def insert_transcript(
    cur,
    *,
    tenant_id: str | UUID,
    voice_note_id: str | UUID,
    text: str,
    provider: str,
    model: str,
    duration_ms: int,
    language: str | None = None,
    cost_usd: float = 0.0,
) -> str:
    cur.execute(
        """
        INSERT INTO transcripts
            (tenant_id, voice_note_id, text, language, provider, model, cost_usd, duration_ms)
        VALUES (%s, %s, %s, %s, %s, %s, %s, %s)
        RETURNING id
        """,
        (tenant_id, voice_note_id, text, language, provider, model, cost_usd, duration_ms),
    )
    return returned_id(cur.fetchone())


def get_transcript(
    cur, *, transcript_id: str | UUID, tenant_id: str | UUID
) -> dict[str, Any]:
    cur.execute(
        """
        SELECT id, voice_note_id, text, language, provider, model
        FROM transcripts
        WHERE id = %s AND tenant_id = %s
        """,
        (transcript_id, tenant_id),
    )
    row = cur.fetchone()
    if row is None:
        raise ValueError(f"transcript not found: {transcript_id}")
    return dict(row)

File: apps/api/internal/database/transcripts.go
package database

import (
	"context"

	"github.com/google/uuid"
)

// InsertTranscriptParams creates a transcript row directly (no audio upload /
// transcribe step). Used by voice intake, where the live conversation already
// produced text that the existing extract job can turn into a draft ticket.
type InsertTranscriptParams struct {
	TenantID    uuid.UUID
	VoiceNoteID uuid.UUID
	Text        string
	Provider    string
	Model       string
}

func InsertTranscript(ctx context.Context, db *DB, p InsertTranscriptParams) (uuid.UUID, error) {
	const q = `
		INSERT INTO transcripts (tenant_id, voice_note_id, text, provider, model)
		VALUES ($1, $2, $3, $4, $5)
		RETURNING id
	`
	var id uuid.UUID
	err := db.QueryRow(ctx, q, p.TenantID, p.VoiceNoteID, p.Text, p.Provider, p.Model).Scan(&id)
	return id, err
}
